website/versioned_docs/version-5.x/server-side-rendering-overview.md

```diff
@@ -0,0 +1,294 @@
+      });
+    },
+    renderFragment(name) {
+      // not relevant to the docs here
+    }
+  });
+
+  bodyStream.pipe(res);
+}).listen(9000)
+```
+
+## 3. HTTP Response Headers
+
+The HTTP response headers sent to the browser are a combination of default headers and the headers retrieved from each microfrontend. Your [method of fetching microfrontends](#2-fetch) does not change how the final headers are merged and assembled for the browser.
+
+Tailor and TailorX have default methods of merging headers. At the time of this writing, single-spa-layout requires manual merging of the headers.
+
+```js
+import {
+  constructServerLayout,
+  setResponseHeaders,
+  renderServerResponseBody,
+} from "single-spa-layout/server";
+import http from 'http';
+
+const serverLayout = constructServerLayout({
+  filePath: "server/views/index.html",
+});
+
+http.createServer((req, res) => {
+  const { bodyStream, applicationProps } = renderServerResponseBody({...});
+
+  const defaultHeaders = {
+    'content-type': 'text/html',
+  }
+
+  setResponseHeaders({
+    res,
+    applicationProps,
+    retrieveApplicationHeaders(props) {
+      return import(`${props.name}/server.js`).then(app => {
+        return app.getHTTPHeaders(props)
+      })
+    },
+    mergeHeaders(headersList) {
+      // This is where you customize which headers to send to the browser
+      const finalHeaders = {...defaultHeaders}
+      headersList.forEach(headers => {
+        for (let headerName in headers) {
+          finalHeaders[headerName] = headers[headerName];
+        }
+      })
+    },
+  }).then(() => {
+    bodyStream.pipe(res);
+  })
+}).listen(9000)
+```
+
+## 4. HTTP Response Body
+
+The HTTP Response body sent from the web server to the browser must be streamed, byte by byte, in order to maximize performance. [NodeJS Readable streams](https://nodejs.org/api/stream.html#stream_readable_streams) make this possible by acting as a buffer that sends each byte as received, instead of all bytes at once.
+
+All microfrontend layout middlewares mentioned in this document stream the HTML response body to the browser. In the context of single-spa-layout, this is done by calling `bodyStream.pipe(res)`
+
+```js
+import {
+  renderServerResponseBody,
+} from "single-spa-layout/server";
+import http from 'http';
+
+const serverLayout = constructServerLayout({
+  filePath: "server/views/index.html",
+});
+
+http.createServer((req, res) => {
+  const { bodyStream, applicationProps } = renderServerResponseBody({...});
+
+  setResponseHeaders({...}).then(() => {
+    bodyStream.pipe(res);
+  })
+}).listen(9000)
+```
+
+## 5. Hydrate
+
+[Helpful article about hydration](https://developers.google.com/web/updates/2019/02/rendering-on-the-web)
+
+Hydration (or rehydration) refers to browser Javascript initializing and attaching event listeners to the HTML sent by the server. There are several variants, including progressive rehydration and partial rehydration.
```

Comment:
I think this one line description is very good, so I'd put the link below it and as an aside.

```suggestion
Hydration (or rehydration) refers to browser Javascript initializing and attaching event listeners to the HTML sent by the server. There are several variants, including progressive rehydration and partial rehydration.

> See also ["Rendering on the Web"](https://developers.google.com/web/updates/2019/02/rendering-on-the-web) by Google.
```